src/app/handle_time.rs

```diff
@@ -0,0 +1,48 @@
+use super::metric_labels::Direction;
+use metrics::{FmtMetrics, Metric};
+use proxy::http::metrics::handle_time;
+use std::{fmt, iter};
+
+#[derive(Clone, Debug)]
+pub struct Metrics {
+    inbound: handle_time::Scope,
+    outbound: handle_time::Scope,
+}
+
+impl Metrics {
+    pub const HELP: &'static str =
+        "A histogram of the time in microseconds between when a request is received and when it is sent upstream.";
+    pub const NAME: &'static str = "request_handle_us";
+
+    pub fn new() -> Self {
+        Self {
+            inbound: handle_time::Scope::new(),
+            outbound: handle_time::Scope::new(),
+        }
+    }
+
+    pub fn outbound(&self) -> handle_time::Scope {
+        self.outbound.clone()
+    }
+
+    pub fn inbound(&self) -> handle_time::Scope {
+        self.inbound.clone()
+    }
```

Comment:
Similarly, I'd be inclined to make a constructor that returns a 3-tuple rather than to hide the scopes in this struct.

Reply:
I'd be happy to change all this code to return tuples, but I'm not sure if it's the right thing in this specific case. I think using tuples as return types is fine when the types in the tuple are heterogeneous (the compiler will notice if you try to call a `Sender` method on a `Receiver`, for example), but I'm a little leery of relying on tuple ordering to differentiate two different instances of the same type, like inbound/outbound.